src/vm.rs

```diff
@@ -136,7 +71,59 @@ pub trait VirtualCPU {
 
 	fn args(&self) -> &[OsString];
 
-	fn cmdsize(&self, syssize: &mut SysCmdsize) {
+	/// `addr` is the address of the hypercall parameter in the guest's memory space. `data` is the
+	/// parameter that was send to that address by the guest.
+	fn address_to_hypercall(&self, addr: u16, data: usize) -> Option<Hypercall<'_>> {
+		if let Ok(hypercall_port) = HypercallAddress::try_from(addr) {
+			Some(match hypercall_port {
+				HypercallAddress::FileClose => {
+					let sysclose = unsafe { &mut *(self.host_address(data) as *mut CloseParams) };
+					Hypercall::FileClose(sysclose)
+				}
+				HypercallAddress::FileLseek => {
+					let syslseek = unsafe { &mut *(self.host_address(data) as *mut LseekParams) };
+					Hypercall::FileLseek(syslseek)
+				}
+				HypercallAddress::FileOpen => {
+					let sysopen = unsafe { &mut *(self.host_address(data) as *mut OpenParams) };
+					Hypercall::FileOpen(sysopen)
+				}
+				HypercallAddress::FileRead => {
+					let sysread = unsafe { &mut *(self.host_address(data) as *mut ReadPrams) };
+					Hypercall::FileRead(sysread)
+				}
+				HypercallAddress::FileWrite => {
+					let syswrite = unsafe { &*(self.host_address(data) as *const WriteParams) };
+					Hypercall::FileWrite(syswrite)
+				}
+				HypercallAddress::FileUnlink => {
+					let sysunlink = unsafe { &mut *(self.host_address(data) as *mut UnlinkParams) };
+					Hypercall::FileUnlink(sysunlink)
+				}
+				HypercallAddress::Exit => {
+					let sysexit = unsafe { &*(self.host_address(data) as *const ExitParams) };
+					Hypercall::Exit(sysexit)
+				}
+				//HypercallPorts::Netwrite => {}
+				//HypercallPorts::Netread => {}
+				//HypercallPorts::Netstat => {}
```

Comment:
We should either include them or not, I think. Rather not, if they are unusued.

```suggestion
```

Reply:
I wanted to remove this together with https://github.com/hermitcore/uhyve/issues/454 in a later PR.
But we can leave it out now as well.